src/pub_verif_token.ts

```diff
@@ -150,57 +216,132 @@ export class TokenResponse {
     }
 }
 
-export function verifyToken(
-    blindRSAMode: BlindRSAMode,
-    token: Token,
-    publicKeyIssuer: CryptoKey,
-): Promise<boolean> {
-    return crypto.subtle.verify(
-        {
-            ...BLIND_RSA.rsaParams,
-            saltLength: blindRSAMode,
-        },
-        publicKeyIssuer,
-        token.authenticator,
-        token.authInput.serialize(),
-    );
-}
+abstract class PubliclyVerifiableIssuer {
+    private suite: (extensions?: Extensions) => Pick<BlindRSA, 'blindSign' | 'verify'>;
 
-export class Issuer {
     constructor(
-        public readonly mode: BlindRSAMode,
+        public readonly mode: BlindRSAMode | PartiallyBlindRSAMode,
         public readonly name: string,
         private readonly privateKey: CryptoKey,
         public readonly publicKey: CryptoKey,
         public readonly params?: BlindRSAPlatformParams,
-    ) {}
+    ) {
+        this.suite = (extensions?: Extensions) => {
+            if (extensions === undefined) {
+                return BLIND_RSA.suite[this.mode]();
+            } else {
+                const suite = PARTIALLY_BLIND_RSA.suite[this.mode]();
+                const serializedExtensions = extensions.serialize();
+                return {
+                    blindSign: (privateKey: CryptoKey, blindMsg: Uint8Array) =>
+                        suite.blindSign(privateKey, blindMsg, serializedExtensions),
+                    verify: (publicKey: CryptoKey, signature: Uint8Array, message: Uint8Array) =>
+                        suite.verify(publicKey, signature, message, serializedExtensions),
+                };
+            }
+        };
+    }
 
-    async issue(tokReq: TokenRequest): Promise<TokenResponse> {
-        const suite = BLIND_RSA.suite[this.mode](this.params);
-        const blindSig = await suite.blindSign(this.privateKey, tokReq.blindedMsg);
+    protected async _issue(tokReq: TokenRequest, extensions?: Extensions): Promise<TokenResponse> {
+        const blindSig = await this.suite(extensions).blindSign(this.privateKey, tokReq.blindedMsg);
         return new TokenResponse(blindSig);
     }
 
     verify(token: Token): Promise<boolean> {
-        return verifyToken(this.mode, token, this.publicKey);
+        return this.suite().verify(
+            this.publicKey,
+            token.authenticator,
+            token.authInput.serialize(),
+        );
+    }
+}
+
+export class Issuer extends PubliclyVerifiableIssuer {
+    constructor(
+        mode: BlindRSAMode | PartiallyBlindRSAMode,
+        name: string,
+        privateKey: CryptoKey,
+        publicKey: CryptoKey,
+        params?: BlindRSAPlatformParams,
+    ) {
+        super(mode, name, privateKey, publicKey, params);
+    }
+
+    async issue(tokReq: TokenRequest): Promise<TokenResponse> {
+        return super._issue(tokReq);
+    }
```

Comment:
also you can make this method public in the abstract class, alternatively you can store the extension in the instance of the class

Reply:
The method has a different signature (due to types) for Issuer with metadata and issuer without.
Therefore the `_issue` method is private, and re-exported as `issue(TokenRequest|ExtendedTokenRequest)` where appropriate.